src/main/kotlin/org/rust/ide/typing/paste/JsonStructParser.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.typing.paste
+
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.core.JsonToken
+import java.io.IOException
+
+sealed class DataType {
+    object Integer : DataType()
+    object Float : DataType()
+    object Boolean : DataType()
+    object String : DataType()
+    object Unknown : DataType()
+    object Missing : DataType()
+    data class StructRef(val struct: Struct) : DataType()
+    data class Array(val type: DataType) : DataType()
+}
+
+data class Field(val key: String, val type: DataType)
+
+class Struct(val fields: List<Field>) {
```

Comment:
Probably it would be easier to use something like
```kotlin
data class Struct(val fields: Map<String, DataType>)
```
or
```kotlin
typealias Struct = Map<String, DataType>
```

Reply:
I want to keep the insertion order of the fields though, that won't be guaranteed with a `Map`.

Reply:
What about `LinkedHashMap`?

Reply:
I switched to it in the latest commit.